javascript/ql/src/semmle/javascript/security/performance/ReDoSUtil.qll

```diff
@@ -802,12 +802,18 @@ private module PrefixConstruction {
       result = prefix(prev) and delta(prev, Epsilon(), state)
       or
       not delta(prev, Epsilon(), state) and
-      result =
-        prefix(prev) +
-          min(string c | delta(prev, any(InputSymbol symbol | c = intersect(Any(), symbol)), state))
+      result = prefix(prev) + getMinimumEdgeChar(prev, state)
     )
   }
 
+  /**
+   * Gets the minimum char for which there exists a transition from `prev` to `next` in the NFA.
+   */
+  private string getMinimumEdgeChar(State prev, State next) {
```

Comment:
Minor: could we hide the "mininum" implementation detail, and simply call this something like `getCanonicalEdgeChar` instead (also update qldoc)?